large-file-uploader-app/assets/iframe.html

```diff
@@ -1,43 +1,48 @@
-	<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@zendeskgarden/css-bedrock@7.0.24/dist/index.min.css">
-	<script type="text/javascript" src="https://assets.zendesk.com/apps/sdk/2.0/zaf_sdk.js"></script>
-</head>
-<body>
-	<script type="text/javascript">
-		var client = ZAFClient.init();
+	<head>
+		<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@zendeskgarden/css-bedrock@7.0.24/dist/index.min.css">
+		<script type="text/javascript" src="https://assets.zendesk.com/apps/sdk/2.0/zaf_sdk.js"></script>
+	</head>
 
-		client.on(
-			'app.registered',
-			function() {
-				client.invoke(
-					'resize',
-					{
-							height: '35px',
-							width: '100%'
-					}
-				);
+	<body>
+		<div id="largeFileUploader"></div>
 
-				client.get('ticket.customField:custom_field_360013377572').then(
-					function(data) {
-						var accountEntryId = data['ticket.customField:custom_field_360013377572']
+		<script type="text/javascript">
+			const client = ZAFClient.init();
 
-						if (accountEntryId && accountEntryId !== '') {
-							client.get('ticket.id').then(
-								function(data) {
-									var element = document.getElementById('largeFileUploader');
-									element.innerHTML = '<a href="https://customer.liferay.com/project-details?p_p_id=com_liferay_osb_customer_account_entry_details_web_AccountEntryDetailsPortlet&p_p_lifecycle=0&p_p_state=normal&p_p_mode=view&_com_liferay_osb_customer_account_entry_details_web_AccountEntryDetailsPortlet_mvcRenderCommandName=%2Fadd_ticket_attachment&_com_liferay_osb_customer_account_entry_details_web_AccountEntryDetailsPortlet_zendeskTicketId=' + data['ticket.id'] + '" target="_blank">Add Large Files over 20 MB here</a>';
-								}
-							);
+			client.on(
+				'app.registered',
+				function() {
+					client.invoke(
+						'resize',
+						{
+								height: '35px',
+								width: '100%'
 						}
-					}
-				).catch(function(error) {
-					console.log(error.toString());
-				});
-			}
-		);
-	</script>
+					);
+
+					Promise.all([client.get('ticket.id'), client.get('ticket.organization')])
+						.then(
+							function(values) {
+								const ticket = values[0];
+								const organization = values[1];
+
+								const organizationObj = organization['ticket.organization'];
 
-	<div id="largeFileUploader"></div>
-</body>
+								if (organizationObj && organizationObj.externalId && organizationObj.id) {
+									const element = document.getElementById('largeFileUploader');
+
+									element.innerHTML = '<a href="https://customer.liferay.com/project-details?p_p_id=com_liferay_osb_customer_account_entry_details_web_AccountEntryDetailsPortlet&p_p_lifecycle=0&p_p_state=normal&p_p_mode=view&_com_liferay_osb_customer_account_entry_details_web_AccountEntryDetailsPortlet_mvcRenderCommandName=%2Fadd_ticket_attachment&_com_liferay_osb_customer_account_entry_details_web_AccountEntryDetailsPortlet_zendeskTicketId=' + ticket['ticket.id'] + '" target="_blank">Add Large Files over 20 MB here</a>';
```

Comment:
Just asking if we should break this up onto multiple lines for better readability?  I tried to do it and in my opinion, it did not make it much more readable since the URL is so long, so my other thought was to make that URL a constant that can be changed elsewhere, but that did not seem to make sense either since the ticket ID is required for the URL to function.  I am open to suggestions to make it better, but if this is the best, then we can keep it.

Let me know your thoughts.  Thanks!

Reply:
I thought about breaking the url into base url `https://customer.liferay.com/` and url params but it's not necessarily easier to read with the additional string concatenation necessary to piece it together. Since this link is unlikely to change, I think it might be fine to leave it as is.

Reply:
However, I have https://issues.liferay.com/browse/LHC-494 open. Once backend implements a friendly url, we could simplify it in both places.

Reply:
Sounds good.  thanks @rdai10!

Reply:
What if we move this whole script to an external JS file?

Reply:
What benefit does an external JS file provide here?